DMCompiler/DMStandard/_Standard.dm

```diff
@@ -133,29 +133,91 @@ proc/block(var/atom/Start, var/atom/End)
 
 	return atoms
 
-proc/range(Dist, atom/Center = usr)
+// TODO: Investigate "for(var/turf/T in range(Dist, Center))"-style weirdness that BYOND does. It's a center-out spiral and we need to replicate that.
+proc/range(Dist, Center)
 	. = list()
 
-	if (isnull(Center)) return
+	var/TrueDist
+	var/atom/TrueCenter
 
-	for (var/x = Center.x - Dist; x <= Center.x + Dist; x++)
-		for (var/y = Center.y - Dist; y <= Center.y + Dist; y++)
-			var/turf/t = locate(x, y, Center.z)
+	if(isnum(Dist))
+		if(isnum(Center))
+			. += Center
+			return
+		if(isnull(Center))
+			TrueCenter = usr
+		else
+			TrueCenter = Center
+		if(isnull(TrueCenter))
+			return
+		if(TrueCenter != usr && !istype(Center, /atom))
+			. += Center
+			return
+		TrueDist = Dist
+	else
+		if(isnull(Center))
```

Comment:
Something like `range(locate(3, 3, 1))` appears to have an infinite range (every object on the map, I have not tested this with z-levels)